src/pip/_internal/commands/search.py

```diff
@@ -76,15 +73,16 @@ def search(self, query, options):
         session = self.get_default_session(options)
 
         transport = PipXmlrpcTransport(index_url, session)
-        pypi = xmlrpc_client.ServerProxy(index_url, transport)
+        pypi = xmlrpc.client.ServerProxy(index_url, transport)
         try:
             hits = pypi.search({'name': query, 'summary': query}, 'or')
-        except xmlrpc_client.Fault as fault:
+        except xmlrpc.client.Fault as fault:
             message = "XMLRPC request failed [code: {code}]\n{string}".format(
                 code=fault.faultCode,
                 string=fault.faultString,
             )
             raise CommandError(message)
+        assert isinstance(hits, list)
```

Comment:
Why is this needed?

Reply:
This assert resolves the error:

```
src/pip/_internal/commands/search.py:85: error:
Incompatible return value type (got "Union[None, bool, int, float, str, bytes, Tuple[Any, ...], List[Any], Dict[Any, Any], datetime, DateTime, Binary]", expected "List[Dict[str, str]]")
            return hits
                   ^
```

The call through `xmlrpc.client` can return any `_Marshallable` type, but here the pip code assumes only a list will be returned.

https://github.com/python/typeshed/blob/a0cc2f537ddd7d6fec9ee3fd46bef6994e44aadd/stdlib/3/xmlrpc/client.pyi#L16